pkg/scheduler/event_handler.go

```diff
@@ -187,23 +187,86 @@ func (s *Scheduler) addCluster(obj interface{}) {
 		return
 	}
 	klog.V(3).Infof("Add event for cluster %s", cluster.Name)
-
+	s.enqueueAffectedPolicy(cluster)
+	s.enqueueAffectedClusterPolicy(cluster)
 	if s.enableSchedulerEstimator {
 		s.schedulerEstimatorWorker.Add(cluster.Name)
 	}
 }
 
-func (s *Scheduler) updateCluster(_, newObj interface{}) {
+func (s *Scheduler) updateCluster(oldObj, newObj interface{}) {
 	newCluster, ok := newObj.(*clusterv1alpha1.Cluster)
 	if !ok {
 		klog.Errorf("cannot convert newObj to Cluster: %v", newObj)
 		return
 	}
+	oldCluster, ok := oldObj.(*clusterv1alpha1.Cluster)
+	if !ok {
+		klog.Errorf("cannot convert oldObj to Cluster: %v", newObj)
+		return
+	}
 	klog.V(3).Infof("Update event for cluster %s", newCluster.Name)
 
 	if s.enableSchedulerEstimator {
 		s.schedulerEstimatorWorker.Add(newCluster.Name)
 	}
+
+	switch {
+	case !equality.Semantic.DeepEqual(oldCluster.Labels, newCluster.Labels):
+		fallthrough
+	case !equality.Semantic.DeepEqual(oldCluster.Spec, newCluster.Spec):
+		s.enqueueAffectedPolicy(newCluster)
+		s.enqueueAffectedClusterPolicy(newCluster)
+	}
```

Comment:
```suggestion
	switch {
	case !equality.Semantic.DeepEqual(oldCluster.Labels, newCluster.Labels):
		fallthrough
	case !equality.Semantic.DeepEqual(oldCluster.Spec, newCluster.Spec):
		fallthrough
	// When cluster becomes 'Ready', re-schedule all relevant RB/CRB
	case !meta.IsStatusConditionTrue(oldCluster.Status.Conditions, clusterv1alpha1.ClusterConditionReady) &&
		meta.IsStatusConditionTrue(newCluster.Status.Conditions, clusterv1alpha1.ClusterConditionReady):
		s.enqueueAffectedPolicy(newCluster)
		s.enqueueAffectedClusterPolicy(newCluster)
	}
```

I guess this probably works. How do you say? @chaunceyjiang

Reply:
I stills think this is a taint changed and should not judge the cluster  status directly.